[Sponsor Page] [Populated] Past Sponsors

CarsonKiibi opened the pull request:
-modified tailwind config ;( (reference i used said it wasnt possible to not) 
-Multiple images copy pasted is part of the trick and on purpose. 
-going to look to re-implement in framer motion if possible tomorrow but current implementation should also work and function as figma describes

Commit:
-scrolling past sponsors infinite
-modified tailwind config ;( (reference i used said it wasnt possible to not)
-Multiple images copy pasted is part of the trick and on purpose
-I am going to look into framer motion tomorrow to make sure there isn't a better solution and maybe ill be able to put up a new commit before pr review

CarsonKiibi commented:
<img width="756" alt="past_sponsors" src="https://github.com/ubc-biomod/biomod-website/assets/43454698/eb036ad8-8e1f-4e91-86d8-082036ff580a">

daniel-panhead commented on src/components/sponsors/PastSponsors.tsx:
```diff
@@ -1,10 +1,162 @@
+            target="_blank"
+            rel="noopener noreferrer"
+          >
+            <img
```
Nextjs recommends using the <Image /> component provided by nextjs because it automatically optimzes images - also that way the source can just be the image variable instead of image.src

daniel-panhead commented on src/components/sponsors/PastSponsors.tsx:
```diff
@@ -1,10 +1,162 @@
+      </h2>
+      <div className="flex overflow-hidden">
+        <div className="flex space-x-2 ml-2 md:ml-8 md:space-x-8 animate-loop-scroll">
+          <a
```
There must be some way to programatically loop through a list of images to generate this instead of copy pasting it a bunch of times

daniel-panhead commented on src/components/sponsors/PastSponsors.tsx:
```diff
@@ -1,10 +1,162 @@
+          >
+            <img
+              src={(amazonIcon as StaticImageData).src}
+              className="h-12 md:h-14 max-w-none filter grayscale hover:grayscale-0"
```
Small nitpick but you can add some transition properties to make the image transition between grayscale (ex. transition-[grayscale property] duration-100 etc.)

daniel-panhead commented on src/stories/sponsors/PastSponsors.stories.ts:
```diff
@@ -1,26 +1,22 @@
+// export default meta;
 const meta = {
-  title: "Sponsors/Past Sponsors",
+  title: "Example/PastSponsors",
```
Use the page it's under as the section title - so keep Sponsors/PastSponsors

daniel-panhead replied:
```suggestion
  title: "Sponsors/PastSponsors",
```

daniel-panhead commented on src/assets/sponsors/tesla-text.png:
where did you even find this logo lmaoo I've never seen it before

daniel-panhead commented on tailwind.config.ts:
There's a bit of a "jump" in the animation when it reaches the end and loops back to the beginning - not sure why though

https://github.com/ubc-biomod/biomod-website/assets/57362494/345de73c-c858-429e-a3c5-fd86f619570b

daniel-panhead reviewed (changes requested):
Overall looks cool, I'd definitely recommend trying framer motion to get some cool animations

IcedTea2K commented on src/components/sponsors/PastSponsors.tsx:
```diff
@@ -1,10 +1,162 @@
+            rel="noopener noreferrer"
+          >
+            <img
+              src={(googleIcon as StaticImageData).src}
```
We could use it directly instead of casting it to StaticImageData.
```suggestion
              src={googleIcon}
```

IcedTea2K reviewed (changes requested):
I really like the animation. A lot of things I wanted to say have already been addressed by Daniel (e.g., animation skipping, loop through all the logos, use frame motion, etc.)

Commit:
-re-implemented with framer motion
-no stutters (at least on my device)
-if logo sizes need to be changed, im pretty sure the size change needs to be proportional to the x value of animate. Kinda confusing but doing so ensures no stutters

CarsonKiibi commented:
-re implemented using framer motion and taking into account past comments

CarsonKiibi commented:
-also added a little fade affect cuz my reference used it lol and it looks cool, can remove tho

daniel-panhead reviewed (approved):
Looks solid!! Just pls fix the story category

daniel-panhead committed:
Populate past sponsors component

IcedTea2K reviewed (approved):
LGTM 👍
